bitnami/thanos/templates/storegateway/hpa-sharded.yaml

```diff
@@ -0,0 +1,58 @@
+{{- if and .Values.storegateway.enabled .Values.storegateway.autoscaling.enabled .Values.storegateway.sharded.enabled }}
+
+{{- $shards := int 0 }}
+{{- if .Values.storegateway.sharded.hashPartitioning.shards }}
+  {{- $shards = int .Values.storegateway.sharded.hashPartitioning.shards }}
+{{- else }}
+  {{- $shards = len .Values.storegateway.sharded.timePartitioning }}
+{{- end }}
+
+{{- range $index, $_ := until $shards }}
+apiVersion: {{ include "common.capabilities.hpa.apiVersion" ( dict "context" $ ) }}
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ printf "%s-storegateway-%s" (include "common.names.fullname" $) (toString $index) | trunc 63 | trimSuffix "-" }}
+  namespace: {{ .Release.Namespace | quote }}
```

Comment:
As you are looping over a `range` function, the scope is changed and `.Release` & `.Values` references can no longer be resolved. You should use `$.Release` and `$.Values` instead.

See the following examples:

https://github.com/bitnami/charts/blob/2a88f4661df35140dc39c72e7677357de764bcc3/bitnami/thanos/templates/storegateway/service-sharded.yaml#L16

https://github.com/bitnami/charts/blob/2a88f4661df35140dc39c72e7677357de764bcc3/bitnami/thanos/templates/storegateway/service-sharded.yaml#L19

Reply:
I just updated the resource and tested the result was the expected one.